docs/README.md

```diff
@@ -0,0 +1,272 @@
+
+The time-based (V1) variant of UUIDs is the fastest to generate. It uses a monotonic clock and node information to generate UUIDs.
+
+```scala mdoc
+import memeid.UUID
+
+UUID.V1.next
+UUID.V1.next
+UUID.V1.next
+```
+
+### Random (v4)
+
+The cryptographically random variant, equivalent to `java.util.UUID/randomUUID`.
+
+```scala mdoc
+import memeid.UUID
+
+UUID.V4.random
+UUID.V4.random
+UUID.V4.random
+```
+
+### Namespaced (v3, v5)
+
+Namespaced UUIDs are generated from a UUID (namespace) and a hashed value (name), V3 uses MD5 and V5 uses SHA1 hash.
+
+```scala mdoc
+import memeid.UUID
+
+val namespace = UUID.V1.next
+
+UUID.V3(namespace, "my-secret-code")
+UUID.V5(namespace, "my-secret-code")
+```
+
+If you want to hash a custom type, you must provide an implicit `memeid.digest.Digestible` instance. This instance is used to convert your type into a byte array for hashing.
+
+```scala mdoc
+import memeid.digest.Digestible
+
+case class User(firstName: String, lastName: String)
+
+implicit val digestibleUser: Digestible[User] =
+  (u: User) => u.firstName.getBytes ++ u.lastName.getBytes
+
+UUID.V3(namespace, User("Federico", "García Lorca"))
+UUID.V5(namespace, User("Federico", "García Lorca"))
+```
+
+### Semi-sequential, random (SQUUID)
+
+SQUUIDs are a non-standard variaton of V4 UUIDs that are semi-sequential. They incorporate a time-component in their 32 most significant bits to generate UUIDs that don't fragment DB indexes.
+
+```scala mdoc
+import memeid.UUID
+
+UUID.V4.squuid
+UUID.V4.squuid
+UUID.V4.squuid
+```
+
+## Java interoperability
+
+`memeid` proviedes conversion method between `UUID` and `java.util.UUID`
+
+```scala mdoc
+import memeid.UUID
+import memeid.JavaConverters._
+import java.util.{ UUID => JavaUUID }
+
+val j = JavaUUID.randomUUID
+val u = UUID.V4.random
+
+j.asScala
+u.asJava
+```
+
+## Literal syntax
+
+`memeid` provides literal syntax with compile-time verification for UUIDs with the `uuid` interpolator. To use it, add this to your `build.sbt`:
+
+
+```scala
+libraryDependencies += "com.47deg" % "memeid-literal" % "@VERSION@"
+```
+
+We can now create UUIDs with literal syntax by importing `memeid.literal._`
+
+```scala mdoc
+import memeid.literal._
+
+uuid"cb096727-6a82-4abd-bc79-fc92be8c5d88"
+```
+
+Invalid UUID literals will fail at compile time:
+
+```scala mdoc:fail
+import memeid.literal._
+
+uuid"not-a-uuid"
+```
+
+## Integrations
+
+`memeid` provides several modules which integrate with popular third-party libraries. If you see something missing don't hesitate to open an issue or send a patch.
+
+### Doobie
+
+The [Doobie](https://github.com/tpolecat/doobie) integration allows you to use the `UUID` type mapped to your database's UUID type.
+
+```scala
+libraryDependencies += "com.47deg" % "memeid-doobie" % "@VERSION@"
+```
+
+To have the [UUID mappings](https://tpolecat.github.io/doobie/docs/12-Custom-Mappings.html) available in scope you can import `memeid.doobie.implicits`.
+
+```scala mdoc
```

Comment:
This could be separated into three blocks:

```
``scala mdoc:invisible
import cats.effect._

import doobie._
import doobie.implicits._
import doobie.h2.implicits._

import scala.concurrent.ExecutionContext

implicit val cs: ContextShift[IO] = IO.contextShift(ExecutionContext.global)

val transactor: Transactor[IO] =
  Transactor.fromDriverManager[IO](
    driver = "org.h2.Driver",
    url = "jdbc:h2:mem:test;DB_CLOSE_DELAY=-1",
    user = "",
    pass = ""
  )

// Create table
sql"CREATE TABLE IF NOT EXISTS test (id UUID NOT NULL)".update.run.transact(transactor).unsafeRunSync
``

``scala mdoc:silent
import memeid.doobie.implicits._

def select(uuid: UUID): Query0[UUID] =
  sql"""SELECT id from test where id = ${uuid}""".query[UUID]

def insert(uuid: UUID): Update0 =
  sql"""insert into test (id) values ($uuid)""".update

val example = UUID.V1.next
``

``scala mdoc
{
    for {
      _ <- insert(example).run.transact(transactor)
      u <- select(example).unique.transact(transactor)
    } yield u
}.unsafeRunSync
``
```

> Notice I added two `` so the block is not broken